### PR TITLE
*: Add few error checks in tests, clean error messages and fix one misspell

### PR DESCRIPTION
**Pull request by dvrkps**

### CODE REVIEW

**Comment by shenli** [comment]
@dvrkps Thanks!
Please make sure you could pass `make dev`.

File: ddl/db_change_test.go [hunk 1]
```diff
@@ -588,6 +588,7 @@ func (s *testStateChangeSuite) TestShowIndex(c *C) {
 	result, err = s.execQuery(tk, "show index from tr;")
 	c.Assert(err, IsNil)
 	err = checkResult(result, testkit.Rows("tr 1 idx1 1 purchased A 0 <nil> <nil>  BTREE  ", "t 1 c2 1 c2 A 0 <nil> <nil> YES BTREE  "))
```
Comment (crazycs520) [inline ddl/db_change_test.go]: the test was wrong.
```go
err = checkResult(result, testkit.Rows("tr 1 idx1 1 purchased A 0 <nil> <nil> YES BTREE  "))
```

File: util/encrypt/crypt_test.go [hunk 2]
```diff
@@ -72,8 +72,8 @@ func (s *testEncryptSuite) TestSQLEncode(c *C) {
 
 	for _, t := range tests {
 		crypted, err := SQLDecode(t.str, t.passwd)
+		c.Assert(err, NotNil)
```
Comment (crazycs520) [inline util/encrypt/crypt_test.go]: c.Assert(err, IsNil)

File: expression/integration_test.go [hunk 3]
```diff
@@ -3127,6 +3127,7 @@ func (s *testIntegrationSuite) TestDateBuiltin(c *C) {
 
 	tk.MustExec("set sql_mode = 'NO_ZERO_DATE'")
 	rs, err := tk.Exec("select date '0000-00-00';")
+	c.Assert(err, NotNil)
```
Comment (crazycs520) [inline expression/integration_test.go]: c.Assert(err, IsNil)

File: expression/integration_test.go [hunk 4]
```diff
@@ -3156,6 +3157,7 @@ func (s *testIntegrationSuite) TestDateBuiltin(c *C) {
 	c.Assert(rs.Close(), IsNil)
 
 	rs, err = tk.Exec("select date '0000-00-00';")
+	c.Assert(err, NotNil)
```
Comment (crazycs520) [inline expression/integration_test.go]: c.Assert(err, IsNil)

**Comment by gregwebs** [comment]
@dvrkps thank you! Lets see if we can get some of these added to the linting/ci now.

File: ddl/db_change_test.go (hunk 1 above)
Comment (dvrkps) [reply ddl/db_change_test.go]: @crazycs520 Done. Thx.

File: util/encrypt/crypt_test.go (hunk 2 above)
Comment (dvrkps) [reply util/encrypt/crypt_test.go]: @crazycs520 Done. Thx.

File: expression/integration_test.go (hunk 3 above)
Comment (dvrkps) [reply expression/integration_test.go]: @crazycs520 Done. Thx.

File: expression/integration_test.go (hunk 4 above)
Comment (dvrkps) [reply expression/integration_test.go]: @crazycs520 Done. Thx.

**Comment by zz-jason** [comment]
@XuHuaiyu @winoros PTAL

**Comment by gregwebs** [comment]
@crazycs520  Can we turn on any new linters in the [Makefile](https://github.com/pingcap/tidb/blob/master/Makefile#L67) now?

**Pull request merged**